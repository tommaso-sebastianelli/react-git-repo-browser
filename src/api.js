import axios from 'axios';
import { from } from 'rxjs';

const CONFIG = {
    API_URL: 'https://api.github.com',
    AUTH_URL: 'https://github.com/login/oauth/authorize',
    CLIENT_ID: '80756f1935ea8314dc97',
    CLIENT_SECRET: 'ed370087dcfc87b2a08e59a3359e1e97a6e24b2a',
    PROXY_URL: 'https://cors-anywhere.herokuapp.com/',
    REDIRECT_URI: 'http://localhost:4200/auth',
    TOKEN_URL: 'https://github.com/login/oauth/access_token',
};

const PATHS = {
    repositories: '/repos'
};

export const listCommits = (owner, repoName) => {
    const url = `${CONFIG.API_URL}${PATHS.repositories}/${owner}/${repoName}/commits`;
    return from(axios
        .get(url, {
          completed: false
        }));
}