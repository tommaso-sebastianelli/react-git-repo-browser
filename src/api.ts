import Observable from 'rxjs/Observable'

const CONFIG = {
    API_URL: 'https://api.github.com',
    AUTH_URL: 'https://github.com/login/oauth/authorize',
    CLIENT_ID: '80756f1935ea8314dc97',
    CLIENT_SECRET: 'ed370087dcfc87b2a08e59a3359e1e97a6e24b2a',
    PROXY_URL: 'https://cors-anywhere.herokuapp.com/',
    REDIRECT_URI: 'http://localhost:4200/auth',
    TOKEN_URL: 'https://github.com/login/oauth/access_token',
};

const PATHS = {
    repositories: '/repos'
  };