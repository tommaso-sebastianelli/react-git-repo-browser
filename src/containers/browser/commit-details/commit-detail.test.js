import React from 'react'
import { Provider } from 'react-redux'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import CommitDetails from './commit-details'
import { getCommit } from '../../../api.mock';
import thunk from 'redux-thunk';
import {
    BrowserRouter as Router,
} from "react-router-dom";

import configureStore from 'redux-mock-store';
const mockStore = configureStore([thunk]);

const getRenderedComponent = async () => {
    const commitData = await getCommit().toPromise();
    const store = mockStore({
        commitReducer: {
            data: commitData
        },
    });

    const component = render(
        <Provider store={store}>
            <Router>
                <CommitDetails />
            </Router>
        </Provider>
    )
    return component;
}


test('render component', async () => {
    const c = await getRenderedComponent();

    expect(c.container).toBeInTheDocument();
})