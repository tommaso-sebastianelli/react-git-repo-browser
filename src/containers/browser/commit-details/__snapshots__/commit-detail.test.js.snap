// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component 1`] = `<UNDEFINED />`;
